Python/pystate.c

```diff
@@ -3002,20 +3002,20 @@ _PyThreadState_CheckConsistency(PyThreadState *tstate)
 //
 // tstate can be a dangling pointer (point to freed memory): only tstate value
 // is used, the pointer is not deferenced.
-//
-// tstate must be non-NULL.
 int
 _PyThreadState_MustExit(PyThreadState *tstate)
 {
+    // tstate must be non-NULL.
+    if (tstate == NULL) {
+        return 1;
+    }
     /* bpo-39877: Access _PyRuntime directly rather than using
        tstate->interp->runtime to support calls from Python daemon threads.
        After Py_Finalize() has been called, tstate can be a dangling pointer:
        point to PyThreadState freed memory. */
     unsigned long finalizing_id = _PyRuntimeState_GetFinalizingID(&_PyRuntime);
     PyThreadState *finalizing = _PyRuntimeState_GetFinalizing(&_PyRuntime);
     if (finalizing == NULL) {
-        // XXX This isn't completely safe from daemon thraeds,
```

Comment:
I would keep this comment, `tstate` could still be dangling in the sense that it's freed memory.